lib/facter/util/partitions/linux.rb

```diff
@@ -55,6 +55,13 @@ def self.filesystem(partition)
       end
     end
     
+    def self.label(partition)
+      label = nil
+      if Facter::Core::Execution.which('blkid')
+        Facter::Core::Execution.exec("blkid #{File.join('/dev', partition)}").scan(/LABEL="([^"]*)"/).flatten.first
```

Comment:
Yes, it doesn't return any key, so it's completely backwards compatible
On 13 Jan 2015 17:40, "Michael Smith" <notifications@github.com> wrote:

> In lib/facter/util/partitions/linux.rb
> <https://github.com/puppetlabs/facter/pull/836#discussion-diff-22874548>:
>
> > @@ -55,6 +55,13 @@ def self.filesystem(partition)
> >        end
> >      end
> >
> > +    def self.label(partition)
> > +      label = nil
> > +      if Facter::Core::Execution.which('blkid')
> > +        Facter::Core::Execution.exec("blkid #{File.join('/dev', partition)}").scan(/LABEL="([^"]*)"/).flatten.first
>
> Have you tried this with an unlabeled partition?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/puppetlabs/facter/pull/836/files#r22874548>.
>